### PR TITLE
Add alpine unittests to ci

### PR DESCRIPTION
**Pull request by holmanb**
## Proposed Commit Message
```
test: Add alpine to ci

Cloud-init has Alpine Linux support, yet does not regularly test it.

Alpine is an attractive addition to the test matrix since it is easily
available to run in lxc, and it has several alternative components to test
against:

- busybox: non-GNU posix-compliant tools
- musl: libc alternative to glibc
- openrc: init system alternative to systemd
```

## Additional Context
~~Blocked by https://github.com/canonical/cloud-init/pull/5119 and https://github.com/canonical/cloud-init/pull/5120. Once those land, unittests will pass on Alpine.~~

Both have landed, ready for review.

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Comment by TheRealFalcon** [comment]
Looks like the job failed:
https://github.com/canonical/cloud-init/actions/runs/8668134702/job/23775704648?pr=5121

**Comment by holmanb** [comment]
> Looks like the job failed:
> https://github.com/canonical/cloud-init/actions/runs/8668134702/job/23775704648?pr=5121

It succeeded a few weeks back but after a rebase to fix a conflict this test started failing and I haven't taken the time to debug it.

File: .github/workflows/alpine-unittests.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+      - name: Check networking (for debugging)
+        run: |
+          sudo sg incus-admin -c 'incus exec alpine -- ping -c 1 google.com || true'
+          sudo sg incus-admin -c 'incus exec alpine -- ping -c 1 dl-cdn.alpinelinux.org || true'
+          sudo sg incus-admin -c 'incus exec alpine -- nslookup www.google.com || true'
+          sudo sg incus-admin -c 'incus exec alpine -- ping -c 1 dl-cdn.alpinelinux.org || true'
+
+      - name: Install dependencies
+        run: sudo sg incus-admin -c 'incus exec alpine -- apk add py3-tox git'
```
Comment (aciba90) [inline .github/workflows/alpine-unittests.yml:58]: > https://github.com/canonical/cloud-init/actions/runs/8668134702/job/23775704648?pr=5121

I could not reproduce the error with the latest image. This sounds to me like an inconsistent state of the repository indexes within the image. Likely avoidable by prepending an `apk update` and optionally an `apk upgrade` before installing `py3-tox`.

**Review by aciba90 — COMMENTED** [review]
Thanks for adding support for unit tests on other platforms than Ubuntu!

Instead of using `incus`, introducing a _new_ tool, what about using the new image server for lxd[1]?

We could add the image server by `lxc remote add images https://images.lxd.canonical.com --protocol=simplestreams` or switch the lxd snap channel to `5.21/stable`, which contains the image server by default.

[1] https://discourse.ubuntu.com/t/new-lxd-image-server-available-images-lxd-canonical-com/43824

**Comment by holmanb** [comment]
> Instead of using incus, introducing a new tool, what about using the new image server for lxd[1]?
>
> We could add the image server by lxc remote add images https://images.lxd.canonical.com --protocol=simplestreams or switch the lxd snap channel to 5.21/stable, which contains the image server by default.

When I proposed this PR, that image server was not yet available to use, but I suppose managing one tool would be simpler than two. I can update it.

**Comment by TheRealFalcon** [comment]
If we're doing LXD (and since the job is still failing), https://github.com/canonical/setup-lxd could be an option too.

**Comment by holmanb** [comment]
@aciba90 @TheRealFalcon I have it working with the lxd Github Actions. Ready for re-review!

**Review by TheRealFalcon — APPROVED** [review]
LGTM!

**Pull request merged**